src/global/project/environment.rs

```diff
@@ -63,3 +80,123 @@ pub struct ParseEnvironmentNameError {
     /// The string that was attempted to be parsed.
     pub attempted_parse: String,
 }
+
+/// Figures out what the status is of the exposed binaries of the environment.
+///
+/// Returns a tuple of the exposed binaries to remove and the exposed binaries to add.
+pub(crate) async fn get_expose_scripts_sync_status(
+    bin_dir: &BinDir,
+    env_dir: &EnvDir,
+    exposed: &IndexMap<ExposedName, String>,
+) -> miette::Result<(IndexSet<PathBuf>, IndexSet<ExposedName>)> {
+    // Get all paths to the binaries from the scripts in the bin directory.
+    let locally_exposed = bin_dir.files().await?;
+    let executable_paths = futures::future::join_all(locally_exposed.iter().map(|path| {
+        let path = path.clone();
+        async move {
+            extract_executable_from_script(&path)
+                .await
+                .ok()
+                .map(|exec| (path, exec))
+        }
+    }))
+    .await
+    .into_iter()
+    .flatten()
+    .collect::<Vec<_>>();
```

Comment:
```suggestion
    .collect_vec();
```

You can import the trait from itertools if not done already